pyproject.toml

```diff
@@ -43,6 +43,7 @@ keywords = [
 requires-python = ">=3.8"
 dependencies = [
     "cmyt>=1.1.2",
+    "h5py>=3.1.0,<4.0.0",
```

Comment:
I think the `<4.0.0` part was probably me trying to sooth my anger for some reason I don't even remember. In hindsight I view this as an anti pattern and I'd rather we remove it if we're going to promote h5py as a hard requirement. You'll also want to add a the corresponding constraint to the `minimal` target.